be/src/storage/olap_server.cpp

```diff
@@ -405,83 +406,19 @@ void* StorageEngine::_pk_index_major_compaction_thread_callback(void* arg) {
-                auto tablet_pk_path = pk_path + "/" + tablet_id;
-                int64_t id = std::stoll(tablet_id);
-                // judge whether tablet should be in the data_dir or not,
-                // for data_dir may change if config:storage_path changed.
-                // just remove if not.
-                if (get_persistent_index_store(id) != data_dir) {
-                    dir_changed_tablet_ids.push_back(id);
-                    if (_clear_persistent_index(data_dir, id, tablet_pk_path).ok()) {
-                        removed_dir_tablet_ids.push_back(id);
-                    }
-                } else if (!lake_tablet_manager->is_tablet_in_worker(id)) {
-                    // the shard may be scheduled to other nodes
-                    if (lake_update_manager->try_lock_pk_index_shard(id)) {
-                        not_in_worker_tablet_ids.emplace_back(id);
-                        // judge whether tablet is scheduled again,
-                        // and pk_index_shard write_lock has been hold, so no process will build the persistent index.
-                        if (!lake_tablet_manager->is_tablet_in_worker(id)) {
-                            // try to remove pk index cache to avoid continuing to use the index in the cache after deletion.
-                            if (lake_update_manager->try_remove_primary_index_cache(id)) {
-                                if (_clear_persistent_index(data_dir, id, tablet_pk_path).ok()) {
-                                    removed_dir_tablet_ids.push_back(id);
-                                }
-                            }
-                        }
-                        lake_update_manager->unlock_pk_index_shard(id);
-                    }
-                }
-            }
-
-            auto debug_vector_info = [](std::vector<int64_t> vector) -> std::string {
-                std::string result;
-                for (int i = 0; i < vector.size(); i++) {
-                    if (i != 0) {
-                        result.append(",");
-                    }
-                    result += std::to_string(vector[i]);
-                }
-                return result;
-            };
-
-            int64_t t_end = MonotonicMillis();
-            LOG(INFO) << "finish gc local persistent index dir: " << pk_path
-                      << ", found tablet not in the worker, tablet_ids: " << debug_vector_info(not_in_worker_tablet_ids)
-                      << ", data_dir changed tablet_ids: " << debug_vector_info(dir_changed_tablet_ids)
-                      << ", and removed dir successfully, tablet_ids: " << debug_vector_info(removed_dir_tablet_ids)
-                      << ", cost:" << t_end - t_start << "ms";
-        }
+        SLEEP_IN_BG_WORKER(config::pindex_shard_data_gc_evict_interval_seconds);
+        lake_update_manager->local_pk_index_mgr()->gc(lake_update_manager);
+        lake_update_manager->local_pk_index_mgr()->evict(lake_update_manager);
```

Comment:
Both `gc()` and `evict()` will call list the same directories, can we eliminate the duplicate list directory operations?

Reply:
done